src/app/directives/has-permission/has-permission.directive.ts

```diff
@@ -0,0 +1,84 @@
+/** Angular Imports */
+import { Directive, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+/** Custom Services */
+import { AuthenticationService } from '../../core/authentication/authentication.service';
+
+/**
+ * Has Permission Directive
+ */
+@Directive({
+  selector: '[mifosxHasPermission]'
+})
+export class HasPermissionDirective {
+
+  /** Describes Visibility */
+  private isVisible: boolean;
+  /** Flag if permission passed is negative */
+  private isNegation = false;
+  /** User Permissions */
+  private userPermissions: any[];
+
+  /**
+   * Extracts User Permissions from User Credentials
+   * @param {TemplateRef} templateRef Template Reference
+   * @param {ViewContainerRef} viewContainer View Container Reference
+   * @param {AuthenticationService} authenticationService AuthenticationService
+   */
+  constructor(private templateRef: TemplateRef<any>,
+              private viewContainer: ViewContainerRef,
+              private authenticationService: AuthenticationService) {
+    const savedCredentials = this.authenticationService.getCredentials();
+    this.userPermissions = savedCredentials.permissions;
+  }
+
+  /**
+   * Evaluates the condition to show template.
+   */
+  @Input()
+  set mifosxHasPermission(permission: any) {
+    if (typeof permission !== 'string') {
+      throw new Error('hasPermission value must be a string');
+    }
+    if (permission.charAt(0) === '!') {
+      this.isNegation = true;
+      permission = permission.slice(1);
+    }
+    if (this.hasPermission(permission) && !this.isNegation) {
+      this.viewContainer.createEmbeddedView(this.templateRef);
+      this.isVisible = true;
+    } else {
+      this.viewContainer.clear();
+      this.isVisible = false;
+    }
+  }
+
+  /**
+   * Checks if user is permitted.
+   * @param {string} permission Permission
+   * @returns {true}
+   * -`ALL_FUNCTIONS`: user is a Super user.
+   * -`ALL_FUNCTIONS_READ`: user has all read permissions and passed permission is 'read' type.
+   * - User has special permission to access that feature.
+   * @returns {false}
+   * - Passed permission doesn't fall under either of above given permission grants.
+   * - Passed permission is null or undefined or incorrectly typed.
+   */
+  private hasPermission(permission: string) {
+    permission = permission.trim();
+    if (this.userPermissions && this.userPermissions.includes('ALL_FUNCTIONS')) {
+      return true;
+    } else if (permission !== null || permission !== undefined) {
```

Comment:
```suggestion
    } else if (permission !== null && permission !== undefined) {
```

Reply:
point, I think we should check only for null string then.